pkg/gen/resources_cmds.go.tpl

```diff
@@ -11,9 +11,11 @@ import (
 )
 
 func addAllResourcesCmds(rootCmd *cobra.Command) {
-	{{ range $specVersion, $vData := .Versions }}{{ if eq $specVersion "v1" }}v1root := rootCmd{{ else }}
+	{{ range $specVersion, $vData := .Versions }}{{ if eq $specVersion "v1" }}v1root := rootCmd{{ else if eq $specVersion "v2-preview" }}
+	previewRoot := resource.NewNamespaceCmd(rootCmd, "preview").Cmd
+	previewV2Root := resource.NewNamespaceCmd(previewRoot, "v2").Cmd{{ else }}
```

Comment:
This is a little hacky but I wanted to ensure that the commands were properly nested under `v2 preview` and this achieves that pretty easily. We may want a more scaleable option in the future if we have v1 previews